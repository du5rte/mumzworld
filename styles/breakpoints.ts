/**
 * Inspired by Luke Wroblewski "ergonomic context" tweet
 * @see http://alexbea.com/blog/ergonomic-breakpoint-variables/
 */
export const breakpoints = {
  any: 0,
  watch: 162,
  watchwide: 184,
  phone: 320,
  phonewide: 390,
  phonewidest: 428,
  tablet: 768,
  laptop: 1024,
  desktop: 1440,
  desktopwide: 1600,
};
