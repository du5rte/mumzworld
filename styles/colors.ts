export const fixedColors = {
  white: '#fff',
  black: '#000',
  transparent: 'transparent',
};

const initialThemeColors = {
  // Primary
  primary: 'hsl(0, 0%, 0%)',
  primaryHover: 'hsl(0, 0%, 17%)',
  primaryPress: 'hsla(0, 0%, 60%, 0.5)',
  // Secondary
  secondary: 'hsl(0, 0%, 30%)',
  secondaryHover: 'hsl(0, 0%, 67%)',
  secondaryPress: 'hsl(0, 0%, 60%)',
  // Tertiary
  tertiary: 'hsl(0, 0%, 68%)',
  // Inactive
  inactive: 'hsl(0, 0%, 60%)',
  inactiveBackground: 'hsl(0, 0%, 95%)',
  // Highlight
  highlight: 'hsl(335, 100%, 40%)',
  highlightHover: 'hsl(335, 100%, 37%)',
  highlightPress: 'hsl(335, 100%, 44%)',
  highlightPressContrast: 'hsl(335, 100%, 76%)',
  // Success
  success: 'hsl(154, 72%, 51%)',
  // Error
  critical: 'hsl(0, 72%, 51%)',
  // Contrast
  contrast: 'hsl(0, 0%, 100%)',
  // Surface
  surface: 'hsl(0, 0%, 96%)',
  // Background
  background: 'hsl(0, 0%, 100%)',
  // Divider
  divider: 'hsl(240, 6%, 90%)',
  dividerHover: 'hsl(240, 5%, 68%)',
  dividerActive: 'hsl(240, 2%, 54%)',
  // Shadows
  shadowColor: 'hsla(0, 0%, 0%, 0.085)',
  shadowColorHover: 'hsla(0, 0%, 0%, 0.04)',
  shadowColorPress: 'hsla(0, 0%, 0%, 0)',
  // skeleton
  skeleton: 'hsl(0, 0%, 77%)',
  skeletonBackground: 'hsl(0, 0%, 88%)',
  // Fixed colors
  ...fixedColors,
};

export type ThemeColors = typeof initialThemeColors;

export const lightThemeColors = initialThemeColors;

export const darkThemeColors: ThemeColors = {
  // Primary
  primary: 'hsl(0, 0%, 100%)',
  primaryHover: 'hsl(0, 0%, 83%)',
  primaryPress: 'hsla(0, 0%, 40%, 0.5)',
  // Secondary
  secondary: 'hsl(0, 0%, 70%)',
  secondaryHover: 'hsl(0, 0%, 33%)',
  secondaryPress: 'hsl(0, 0%, 40%)',
  // Tertiary
  tertiary: 'hsl(0, 0%, 32%)',
  // Inactive
  inactive: 'hsl(0, 0%, 40%)',
  inactiveBackground: 'hsl(0, 0%, 15%)',
  // Highlight
  highlight: 'hsl(335, 100%, 40%)',
  highlightHover: 'hsl(335, 100%, 37%)',
  highlightPress: 'hsl(335, 100%, 44%)',
  highlightPressContrast: 'hsl(335, 100%, 76%)',
  // Success
  success: 'hsl(154, 72%, 51%)',
  // Error
  critical: 'hsl(0, 72%, 51%)',
  // Contrast
  contrast: 'hsl(0, 0%, 10%)',
  // Background
  background: 'hsl(0, 0%, 10%)',
  // Surface
  surface: 'hsl(0, 0%, 14%)',
  // Divider
  divider: 'hsl(240, 6%, 30%)',
  dividerHover: 'hsl(240, 5%, 32%)',
  dividerActive: 'hsl(240, 2%, 46%)',
  // Shadows
  shadowColor: 'hsla(0, 0%, 0%, 0.2)',
  shadowColorHover: 'hsla(0, 0%, 0%, 0.1)',
  shadowColorPress: 'hsla(0, 0%, 0%, 0.05)',
  // Skeleton
  skeleton: 'hsl(0, 0%, 18%)',
  skeletonBackground: 'hsl(0, 0%, 24%)',
  // Fixed colors
  ...fixedColors,
};
