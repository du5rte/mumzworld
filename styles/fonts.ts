import { createFont } from '@tamagui/core';

const fontSizes = {
  1: 12,
  2: 14,
  3: 16,
  4: 18,
  5: 20,
  6: 24,
  7: 28,
  8: 35,
  9: 60,
};

const lineHeights = {
  1: 16,
  2: 20,
  3: 24,
  4: 26,
  5: 28,
  6: 30,
  7: 36,
  8: 40,
  9: 60,
};

const headingFont = createFont({
  family: 'Inter, Helvetica, Arial, sans-serif',
  size: { ...fontSizes },
  lineHeight: { ...lineHeights },
  weight: {
    1: '500',
  },
  face: {
    400: { normal: 'Inter-Regular' },
    500: { normal: 'Inter-Medium' },
    600: { normal: 'Inter-SemiBold' },
  },
});

const bodyFont = createFont({
  family: 'Inter, Helvetica, Arial, sans-serif',
  size: { ...fontSizes },
  lineHeight: { ...lineHeights },
  weight: {
    1: '400',
  },
  face: {
    400: { normal: 'Inter-Regular' },
    500: { normal: 'Inter-Medium' },
    600: { normal: 'Inter-SemiBold' },
  },
});

const monoFont = createFont({
  family: 'NunitoSans, Helvetica, Arial, sans-serif',
  size: { ...fontSizes },
  lineHeight: { ...lineHeights },
  weight: {
    1: '500',
  },
  face: {
    500: { normal: 'NunitoSans-Medium' },
  },
});

export const fonts = {
  heading: headingFont,
  body: bodyFont,
  mono: monoFont,
};
