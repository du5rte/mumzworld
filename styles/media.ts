import { createMedia } from '@tamagui/react-native-media-driver';

/**
 * Inspired by Luke Wroblewski "ergonomic context" tweet
 * @see http://alexbea.com/blog/ergonomic-breakpoint-variables/
 */
export const media = createMedia({
  initial: { minWidth: 0 },
  watch: { minWidth: 162 },
  watchwide: { minWidth: 184 },
  phone: { minWidth: 320 },
  phonewide: { minWidth: 390 },
  phonewidest: { minWidth: 428 },
  tablet: { minWidth: 768 },
  laptop: { minWidth: 1024 },
  desktop: { minWidth: 1440 },
  desktopwide: { minWidth: 1600 },
  hoverNone: { hover: 'none' },
  pointerCoarse: { pointer: 'coarse' },
});
