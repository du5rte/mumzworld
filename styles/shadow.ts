import { Platform } from 'react-native';
import theme from './themes';
import { SharedValue } from 'react-native-reanimated';
import { withEaseOutQuad, withEaseOutSin } from './timings';

// This function is used to generate shadow by elevation
// See more: https://github.com/tamagui/tamagui/blob/master/code/ui/stacks/src/getElevation.tsx
export function generateElevation(amount: number = 10, shadowColor = theme.colors.shadowColor) {
  const [height, shadowRadius] = [Math.round(amount / 4 + 1), Math.round(amount / 2 + 2)];
  const shadow = {
    shadowColor: shadowColor,
    shadowRadius,
    shadowOffset: { height, width: 0 },
    ...(Platform.OS === 'android' && {
      elevationAndroid: 2 * height,
    }),
    ...(Platform.OS === 'web' && {
      boxShadow: `0 ${height}px ${shadowRadius}px ${shadowColor}`,
      transition: 'box-shadow 0.3s ease-out',
    }),
  };
  return shadow;
}

export function interactiveElevationChange(
  amount: number,
  { press, hover }: { press?: SharedValue<boolean>; hover?: SharedValue<boolean> }
) {
  'worklet';

  const height = Math.round(amount / 4 + 1);
  const shadowRadius = Math.round(amount / 2 + 2);
  const elevation = 2 * height;

  return {
    shadowColor: 'hsl(0, 0%, 0%)',
    // Press increases the shadow radius making it seem the light is being diffused
    shadowRadius: press?.value ? withEaseOutQuad(shadowRadius * 2) : withEaseOutSin(shadowRadius),
    // Press shifts the shadow closer to the button making it seem the button is being pressed against the surface
    shadowOffset: {
      height: press?.value ? withEaseOutQuad(0) : withEaseOutSin(height),
      width: 0,
    },
    // Press turns down the opacity making it seem button is being pressed against the surface
    shadowOpacity: press?.value ? withEaseOutQuad(0) : withEaseOutSin(0.085),
    ...(Platform.OS === 'android' && {
      elevation: press?.value ? withEaseOutQuad(0) : withEaseOutSin(elevation),
    }),
    ...(Platform.OS === 'web' && {
      // Hover shifts the shadow away from the button making it seem the button is floating
      // Press shifts the shadow closer to the button making it seem the button is being pressed against the surface
      boxShadow: `0 ${press?.value ? 0 : hover?.value ? height * 2 : height}px ${press?.value ? shadowRadius * 2 : shadowRadius}px ${press?.value ? theme.colors.shadowColorPress : theme.colors.shadowColor}`,
      transition: 'box-shadow 0.3s ease-out',
    }),
  };
}
