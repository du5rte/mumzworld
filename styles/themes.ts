import bottomTabBar from '@/components/bottom-tab-bar';
import { createTheme } from '@shopify/restyle';
import { Platform } from 'react-native';

export const theme = createTheme({
  colors: {
    background: 'hsl(0, 0%, 100%)', // #FFFFFF
    surface: 'hsl(0, 0%, 98%)', // #FAFAFA

    // Primary
    primary: 'hsl(0, 0%, 0%)',
    primaryInvert: 'hsl(0, 0%, 100%)', // #FFFFFF
    primaryHover: 'hsl(0, 0%, 17%)', // #383838
    primaryPress: 'hsla(0, 0%, 60%, 0.5)',
    primaryBackgroundPress: 'hsl(0, 0%, 20%)',

    // Secondary
    secondary: 'hsl(0, 0%, 60%)', // Lightened
    secondaryInvert: 'hsl(0, 0%, 0%)', // Inverted
    secondaryHover: 'hsl(0, 0%, 67%)', // Lightened
    secondaryPress: 'hsl(0, 0%, 60%)', // Lightened

    // Tertiary
    tertiary: 'hsl(0, 0%, 68%)', // #ADADAD

    // Inactive
    inactive: 'hsl(0, 0%, 53%)', // #878787
    inactiveSurface: 'hsl(0, 0%, 95%)',

    // Highlight
    highlight: 'hsl(216, 100%, 48%)',
    highlightHover: 'hsl(216, 100%, 45%)',
    highlightPress: 'hsl(216, 100%, 52%)',
    highlightActive: 'hsl(216, 100%, 57%)',
    highlightInvertPress: 'hsl(216, 100%, 83%)',

    // Success
    success: 'hsl(154, 72%, 51%)',
    successDivider: 'hsl(154, 74%, 76%)',

    // Error
    error: 'hsl(0, 72%, 51%)',
    errorDivider: 'hsl(1, 74%, 76%)',

    // Divider
    divider: 'hsl(240, 6%, 83%)', // #EDEDED blended to #D2D2D7 to transitions easier with highlight
    dividerHover: 'hsl(240, 5%, 68%)', // #ADADAD blended to #A9A9B1 to transitions easier with highlight
    dividerActive: 'hsl(240, 2%, 54%)', // #8C8C8C blended to #86868B to transitions easier with highlight

    // Shadows
    shadowColor: 'hsla(0, 0%, 0%, 0.085)',
    shadowColorHover: 'hsla(0, 0%, 0%, 0.04)',
    shadowColorFocus: 'hsla(0, 0%, 0%, 0.04)',
    shadowColorPress: 'hsla(0, 0%, 0%, 0)',

    // Transparent
    transparent: 'transparent',
  },
  spacing: {
    none: 0,
    xxs: 2,
    xs: 4,
    /**
     * used for in-between items
     * 8px
     */
    s: 8,
    /**
     * used for breathing space
     * 12px
     */
    m: 12,
    /**
     * used for wrapping items (side margins)
     * 16px
     */
    l: 16,
    /**
     * used for line breaks
     * 24px
     */
    xl: 24,
    /**
     * used for give items white space (room)
     * 36px
     */
    '2xl': 36,
  },
  borderRadii: {
    none: 0,
    s: 4,
    m: 8,
    circle: 9999,
  },
  /**
   * Inspired by Luke Wroblewski "ergonomic context" tweet
   * @see http://alexbea.com/blog/ergonomic-breakpoint-variables/
   */
  breakpoints: {
    any: 0,
    watch: 162,
    watchwide: 184,
    phone: 320,
    phonewide: 390,
    phonewidest: 428,
    tablet: 768,
    laptop: 1024,
    desktop: 1440,
    desktopwide: 1600,
  },
  /**
   * Inspired by shopify polaris
   * @see https://polaris.shopify.com/design/typography#section-display-styles
   */
  textVariants: {
    /**
     * Inherit by all
     */
    defaults: {
      fontFamily: 'Inter-Regular',
      ...(Platform.OS === 'web' && {
        fontFamily: 'Inter, sans-serif',
        fontWeight: '400',
      }),
      color: 'primary',
      fontSize: 16,
      lineHeight: 24,
    },
    semiBold: {
      fontFamily: 'Inter-medium',
      ...(Platform.OS === 'web' && {
        fontFamily: 'Inter, sans-serif',
        fontWeight: '500',
      }),
      fontSize: 16,
      lineHeight: 24,
    },
    bold: {
      fontFamily: 'Inter-SemiBold',
      ...(Platform.OS === 'web' && {
        fontFamily: 'Inter, sans-serif',
        fontWeight: '600',
      }),
      fontSize: 16,
      lineHeight: 24,
    },
    title: {
      fontFamily: 'Inter-Bold',
      ...(Platform.OS === 'web' && {
        fontFamily: 'Inter, sans-serif',
        fontWeight: '700',
      }),
      fontSize: 32,
      lineHeight: 32,
    },
    subtitle: {
      fontFamily: 'Inter-medium',
      ...(Platform.OS === 'web' && {
        fontFamily: 'Inter, sans-serif',
        fontWeight: '500',
      }),
      fontSize: 20,
      lineHeight: 24,
    },
  },
});

export type Theme = typeof theme;

export type ThemeColors = keyof Theme['colors'];
export type ThemeBorderRadiuses = keyof Theme['borderRadii'];

export const lightTheme = theme;

export const darkTheme: Theme = {
  ...theme,
  colors: {
    ...theme.colors,
    background: 'hsl(0, 0%, 10%)', // Darkened
    surface: 'hsl(0, 0%, 12%)', // Darkened

    // Dark Theme Colors
    // Primary
    primary: 'hsl(0, 0%, 100%)', // Inverted
    primaryInvert: 'hsl(0, 0%, 0%)', // Inverted
    primaryHover: 'hsl(0, 0%, 83%)', // Lightened
    primaryPress: 'hsl(0, 0%, 100%)', // Inverted
    primaryBackgroundPress: 'hsl(0, 0%, 80%)', // Lightened

    // Secondary
    secondary: 'hsl(0, 0%, 60%)', // Lightened
    secondaryInvert: 'hsl(0, 0%, 100%)', // Inverted
    secondaryHover: 'hsl(0, 0%, 67%)', // Lightened
    secondaryPress: 'hsl(0, 0%, 60%)', // Lightened

    // Tertiary
    tertiary: 'hsl(0, 0%, 32%)', // Darkened

    // Inactive
    inactive: 'hsl(0, 0%, 47%)', // Adjusted
    inactiveSurface: 'hsl(0, 0%, 15%)', // Darkened

    // Highlight
    highlight: 'hsl(216, 100%, 52%)', // Adjusted
    highlightHover: 'hsl(216, 100%, 55%)', // Adjusted
    highlightPress: 'hsl(216, 100%, 48%)', // Adjusted
    highlightActive: 'hsl(216, 100%, 43%)', // Adjusted
    highlightInvertPress: 'hsl(216, 100%, 17%)', // Adjusted

    // Success
    success: 'hsl(154, 72%, 49%)', // Adjusted
    successDivider: 'hsl(154, 74%, 24%)', // Adjusted

    // Error
    error: 'hsl(0, 72%, 49%)', // Adjusted
    errorDivider: 'hsl(1, 74%, 24%)', // Adjusted

    // Divider
    divider: 'hsl(240, 6%, 17%)', // Lightened for contrast
    dividerHover: 'hsl(240, 5%, 32%)', // Lightened for contrast
    dividerActive: 'hsl(240, 2%, 46%)', // Lightened for contrast

    // Shadows
    shadowColor: 'hsla(0, 0%, 0%, 0.2)', // Adjusted opacity
    shadowColorHover: 'hsla(0, 0%, 0%, 0.1)', // Adjusted opacity
    shadowColorFocus: 'hsla(0, 0%, 0%, 0.1)', // Adjusted opacity
    shadowColorPress: 'hsla(0, 0%, 0%, 0.05)', // Adjusted opacity

    // Transparent
    transparent: 'transparent', // Unchanged
  },
};

export const themeVariants = ['light', 'dark'] as const;
export type ThemeVariants = (typeof themeVariants)[number];

export const themes: Record<ThemeVariants, Theme> = {
  light: lightTheme,
  dark: darkTheme,
};

export default theme;
