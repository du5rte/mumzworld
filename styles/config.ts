import { createTamagui } from '@tamagui/core';

import { fonts } from './fonts';
import { tokens } from './tokens';
import { media } from './media';
import { animations } from './animations';
import { lightTheme, darkTheme } from './themes2';

export const config = createTamagui({
  fonts,
  tokens,
  themes: {
    light: lightTheme,
    dark: darkTheme,
  },
  media,
  animations,
  shorthands: {
    f: 'flex',
    m: 'margin',
    w: 'width',
    c: 'color',
    bg: 'backgroundColor',
    br: 'borderRadius',
    bc: 'borderColor',
  } as const,
  settings: {
    allowedStyleValues: 'somewhat-strict',
    autocompleteSpecificTokens: 'except-special',
  },
});
