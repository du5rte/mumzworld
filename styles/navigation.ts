import { Platform, TextStyle, ViewStyle } from 'react-native';
import {
  DefaultTheme as DefaultNavigationTheme,
  DarkTheme as NavigationDarkTheme,
  Theme as NavigationTheme,
} from '@react-navigation/native';
import { darkTheme, lightTheme } from './themes';
import { spaces } from './tokens';

export const headerStyle: ViewStyle = {
  backgroundColor: 'transparent',
  ...Platform.select({
    android: {
      elevation: 0,
    },
    ios: {
      shadowColor: 'transparent',
      shadowOpacity: 0,
      shadowRadius: 0,
      shadowOffset: {
        width: 0,
        height: 0,
      },
    },
    default: {
      // https://github.com/necolas/react-native-web/issues/44
      // Material Design
      boxShadow: 'none',
    },
  }),
};

export const headerTitleStyle: TextStyle = {
  fontFamily: 'Inter-Medium',
  ...(Platform.OS === 'web' && {
    fontFamily: 'Inter, sans-serif',
    fontWeight: '500',
  }),
  fontSize: 17,
  lineHeight: 24,
};

export const headerRightContainerStyle: ViewStyle = {
  paddingRight: spaces.xl,
};

export const lightNavigationTheme: NavigationTheme = {
  ...DefaultNavigationTheme,
  colors: {
    ...DefaultNavigationTheme.colors,
    primary: lightTheme.colors.primary,
    background: lightTheme.colors.background,
  },
};

export const darkNavigationTheme: NavigationTheme = {
  ...NavigationDarkTheme,
  colors: {
    ...NavigationDarkTheme.colors,
    primary: darkTheme.colors.primary,
    background: darkTheme.colors.background,
  },
};
