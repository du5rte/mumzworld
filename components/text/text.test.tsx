import { render, screen } from '@/utils/tests/react-native';
import { ThemeProvider } from '@shopify/restyle';

import Text from './text';
import { themes, themeVariants } from '@/styles/themes';
import { body } from '@/styles/fonts';

describe('text', () => {
  const text = 'Hello World';

  it('should render text correctly', () => {
    render(<Text>{text}</Text>);

    expect(screen.getByText(text)).toBeDefined();
  });

  it.each(themeVariants)(
    'should render default text styles correctly for %p theme',
    (themeVariant) => {
      const theme = themes[themeVariant];

      render(
        <ThemeProvider theme={theme}>
          <Text>{text}</Text>
        </ThemeProvider>
      );

      expect(screen.getByText(text)).toHaveStyle({
        fontSize: body.fontSize,
        lineHeight: body.lineHeight,
        color: theme.colors[body.color],
      });
    }
  );

  it('should render no text styles when using unstyled', () => {
    render(<Text variant="unstyled">{text}</Text>);

    expect(screen.getByText(text)).toHaveStyle({});
  });
});
