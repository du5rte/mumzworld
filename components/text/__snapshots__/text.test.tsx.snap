// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text should render correctly 1`] = `
<Text
  style={
    [
      {
        "color": "hsl(0, 0%, 0%)",
        "fontFamily": "Inter-Regular",
        "fontSize": 16,
        "lineHeight": 24,
      },
    ]
  }
/>
`;
